tools/state-viewer/src/trie_iteration_benchmark.rs

```diff
@@ -0,0 +1,308 @@
+            }
+        }
+
+        for (shard_id, chunk_header) in block.chunks().iter().enumerate() {
+            let shard_id = shard_id as ShardId;
+            if chunk_header.height_included() != block.header().height() {
+                println!("chunk for shard {shard_id} is missing, skipping it");
+                continue;
+            }
+            let trie = self.get_trie(shard_id, &shard_layout, &chunk_header, &store);
+
+            println!("shard id    {shard_id:#?} benchmark starting");
+            self.iter_trie(&trie);
+            println!("shard id    {shard_id:#?} benchmark finished");
+        }
+    }
+
+    fn get_trie(
+        &self,
+        shard_id: ShardId,
+        shard_layout: &ShardLayout,
+        chunk_header: &ShardChunkHeader,
+        store: &Store,
+    ) -> Trie {
+        let shard_uid = ShardUId::from_shard_id_and_layout(shard_id, shard_layout);
+        // Note: here we get the previous state root but the shard layout
+        // corresponds to the current epoch id. In practice shouldn't
+        // matter as the shard layout doesn't change.
+        let state_root = chunk_header.prev_state_root();
+        let storage = TrieDBStorage::new(store.clone(), shard_uid);
+        let flat_storage_chunk_view = None;
+        Trie::new(Rc::new(storage), state_root, flat_storage_chunk_view)
+    }
+
+    fn iter_trie(&self, trie: &Trie) {
+        let stats = Rc::new(RefCell::new(TrieIterationBenchmarkStats::new()));
+        let stats_clone = Rc::clone(&stats);
+
+        let prune_condition: Option<Box<dyn Fn(&Vec<u8>) -> bool>> = match &self.iteration_type {
+            TrieIterationType::Full => None,
+            TrieIterationType::Shallow => Some(Box::new(move |key_nibbles| -> bool {
+                Self::shallow_iter_prune_condition(key_nibbles, &stats_clone)
+            })),
+        };
+
+        let start = Instant::now();
+        let mut node_count = 0;
+        let mut error_count = 0;
+        let iter = trie.iter_with_prune_condition(prune_condition);
+        let iter = match iter {
+            Ok(iter) => iter,
+            Err(err) => {
+                println!("iter error {err:#?}");
+                return;
+            }
+        };
+        for item in iter {
+            node_count += 1;
+
+            let (key, value) = match item {
+                Ok((key, value)) => (key, value),
+                Err(err) => {
+                    println!("Failed to iterate node with error: {err}.");
+                    error_count += 1;
+                    continue;
+                }
+            };
+
+            stats.borrow_mut().bump_visited(key[0]);
+
+            let state_record = StateRecord::from_raw_key_value(key.clone(), value);
+            let state_record = match state_record {
+                Some(state_record) => state_record,
+                None => {
+                    println!("Failed to parse state record.");
```

Comment:
yep, I'll change that to tracing::error
the key and value are not available yet, this is the attempt to unpack the state_record into key and value

Reply:
I'm confused. I see this `None` as a failing option to `StateRecord::from_raw_key_value(key.clone(), value)`.
So, we should be able to write `key` and `value` in logs. The question is do we want to? 

Ah, maybe there is a mixup between this comment and comment for line 182, because I commented out of order.

Reply:
Ah yeah you're right I read it in the comment order, not in the code order :D I do have key and value here. I'll add it if I need it, for now it's fine ;)